Product creation: Update size of bottom sheet based on size class

itsmeichigo opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #10989 
<!-- Id number of the GitHub issue this PR addresses. -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR updates the large detent mode for the product creation bottom sheet:
- Disables large detent if the device is iPad.
- For iPhones, enable both large and medium detents.

📝 I removed the custom height detent because it doesn't work well. It would be nice if we could calculate the height of the sheet before presenting but I'm not sure if that's possible. Please feel free to suggest any improvements.

## Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
- Log in to a WPCom site or self-hosted site with AI assistance available.
- Navigate to the Products tab and select + to add a new product.
- Notice the bottom sheet is displayed properly.
- Repeat with different devices and font sizes.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->
Landscape mode for iPhone: always enable large detent:
![Simulator Screenshot - iPhone 14 Pro - 2023-10-24 at 13 16 15](https://github.com/woocommerce/woocommerce-ios/assets/5533851/84cf5639-3a07-4614-965f-f9b42e0f33af)

Portrait mode for iPhone: enable both large and medium detents:
| Normal font size | Large font size | 
| --- | --- | 
| <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/76e5ffc6-297e-4346-8096-548ac180337f" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/f579069f-2941-423b-a580-fb8b587c825d" width=320 /> |

On iPad: disable large detent.

<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/06acd6ed-073d-4adf-bb42-4eade9728438" width=400 />

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

itsmeichigo committed:
Fix the cut off Learn more text for large font size

itsmeichigo committed:
Update size class for product creation bottom sheet

If preferred content size is accessibility or vertical size class is compact, add large detent for accessibility. Skip if device is iPad.

itsmeichigo committed:
Remove magic size for bottom sheet

itsmeichigo committed:
Remove outdated comments

selanthiraiyan commented on WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift:
```diff
@@ -393,22 +393,27 @@ private extension AddProductCoordinator {
             var sheet = bottomSheet
             sheet.prefersEdgeAttachedInCompactHeight = true
             sheet.largestUndimmedDetentIdentifier = .none
             sheet.prefersGrabberVisible = true
```
❓ Would it make sense to hide the grabber when on iPad? 

<img src="https://github.com/woocommerce/woocommerce-ios/assets/524475/9e6692d0-ceeb-4d50-a838-a017f00591e5" width="500"/>

itsmeichigo replied:
Makes sense! Removed in 9a707551192e0ddc11f6f502501501efd2082580.

selanthiraiyan reviewed (commented):
I tested this on an iPhone 14 Pro simulator running iOS 17. Xcode Version 14.3.1.

A crash occurs upon rotating the simulator. 

```
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'At least one detent must be active.'
```

itsmeichigo committed:
Keep both medium and large detents for iphones

itsmeichigo committed:
Hide grabber for iPad

itsmeichigo commented:
> A crash occurs upon rotating the simulator.

Thank you @selanthiraiyan for catching the crash! Looks like having only medium detent on portrait mode causes a crash when transitioning to landscape mode. I updated the logic in e28c5ee4987a2342a8a2323af5dd529db4603f00 to enable both large and medium detents for iPhone. I also updated the description of the PR accordingly. Please take another look when you can.

selanthiraiyan reviewed (approved):
Thanks for the updates, Huong! 🚀
